game-core/src/test/java/games/strategy/triplea/attachments/TriggerAttachmentTest.java

```diff
@@ -26,102 +29,226 @@
+      final TriggerAttachment triggerAttachment = mock(TriggerAttachment.class);
+      final Set<TriggerAttachment> satisfiedTriggers = Collections.singleton(triggerAttachment);
+
+      when(triggerAttachment.getPropertyMap()).thenCallRealMethod();
+      triggerAttachment.getPropertyMap().get("playerAttachmentName").setValue("rulesAttachment:RulesAttachment");
+      when(triggerAttachment.getPlayerProperty())
+          .thenReturn(Collections.singletonList(
+              Tuple.of("productionPerXTerritories", "someNewValue")));
+      when(triggerAttachment.getName()).thenReturn("mockedTriggerAttachment");
+
+      final PlayerId playerId = mock(PlayerId.class);
+      when(playerId.getAttachment("rulesAttachment"))
+          .thenReturn(new RulesAttachment(null, null, gameData));
+      when(triggerAttachment.getPlayers()).thenReturn(Collections.singletonList(playerId));
+
+      TriggerAttachment.triggerPlayerPropertyChange(
+          satisfiedTriggers,
+          bridge,
+          "beforeOrAfter",
+          "stepName",
+          false, // useUses
+          false, // testUses
+          false, // testChance
+          false); // testWhen
+      verify(bridge).addChange(not(argThat(Change::isEmpty)));
+    }
+
+    // TODO: Any existing games using 'triggerRelationshipTypePropertyChange', to base the test parameters on?
+
+    @Test
+    void testTriggerTerritoryPropertyChange() throws Exception {
+      final GameData gameData = bridge.getData();
+      final TriggerAttachment triggerAttachment = mock(TriggerAttachment.class);
+      final Set<TriggerAttachment> satisfiedTriggers = Collections.singleton(triggerAttachment);
+
+      when(triggerAttachment.getPropertyMap()).thenCallRealMethod();
+      triggerAttachment.getPropertyMap().get("territoryAttachmentName")
+          .setValue("territoryAttachment:TerritoryAttachment");
+      when(triggerAttachment.getTerritoryProperty())
+          .thenReturn(Collections.singletonList(
+              Tuple.of("kamikazeZone", "true")));
+      when(triggerAttachment.getName()).thenReturn("mockedTriggerAttachment");
+
+      final Territory territory = mock(Territory.class);
+      when(territory.getAttachment("territoryAttachment"))
+          .thenReturn(new TerritoryAttachment(null, null, gameData));
+      when(triggerAttachment.getTerritories()).thenReturn(Collections.singletonList(territory));
+
+      TriggerAttachment.triggerTerritoryPropertyChange(
+          satisfiedTriggers,
+          bridge,
+          "beforeOrAfter",
+          "stepName",
+          false, // useUses
+          false, // testUses
+          false, // testChance
+          false); // testWhen
+      verify(bridge).addChange(not(argThat(Change::isEmpty)));
+    }
+
+    // TODO: Any existing games using 'triggerTerritoryEffectPropertyChange', to base the test parameters on?
+
+    // TODO: Any existing games using 'triggerUnitPropertyChange', to base the test parameters on?
```

Comment:
Doesn't look like those are being used either @ron-murhammer

Reply:
https://github.com/search?q=org%3Atriplea-maps+UnitProperty&type=Code

Reply:
Thanks and I apologize, I think I messed something up when I searched for it in the existing map I have cloned and search in. I have refactored it and added a unit test.